### ISSUE TITLE
Do we verify the Signature?

### ISSUE THREAD

**Issue by gavinharris-dev** [pyropy/web3-cardano-token#2]
Does the code currently verify the authenticity of the signed message in any way? I can see that we are ensuring that the token is correctly formed but are we checking if the signature provided is a signature for this token?

**Comment by pyropy** [comment]
No there is no code for signature checking currently in place.

**Comment by gavinharris-dev** [comment]
Okay cool; I will investigate as this would be important to try preventing spoofing of a token

**Comment by pierre-andre-long** [comment]
Thanks for this update, i implemented it and it works fine until I changed the implementation of the cardano api call to the new one, check #5 .

**Comment by gavinharris-dev** [comment]
Yeah I need to rework the way that I am integrating with the Wallet; CIP-30 changed this process to be more generic.

**Comment by gavinharris-dev** [comment]
I've updated my fork to handle the updated 'signData' api response. It was a little different to just returning a 'string' output.
